backend/dataall/modules/s3_datasets/db/dataset_repositories.py

```diff
@@ -183,6 +189,20 @@ def get_dataset_tables(session, dataset_uri):
         """return the dataset tables"""
         return session.query(DatasetTable).filter(DatasetTable.datasetUri == dataset_uri).all()
 
+    @staticmethod
+    def update_dataset_shares_expiration(session, dataset):
+        """
+        When share expiration is enabled on the dataset while editing a dataset
+        update all the shares on that dataset and set minimum expiration on them
+        """
+        shares = session.query(ShareObject).filter(ShareObject.datasetUri == dataset.datasetUri).all()
+        for share in shares:
+            share.expiryDate = ShareObjectService.calculate_expiry_date(
+                expirySetting=dataset.expirySetting, expirationPeriod=dataset.expiryMinDuration
```

Comment:
What happens, if
1. Current expiration is less then new `dataset.expiryMinDuration`?
2. Current expiration is more then new `dataset.expiryMaxDuration`?

Reply:
Hmm. That's a good point. I will incorporate some checks 
I am thinking 
if the share already has an expiration date which is present, then I won't change the expiration period, but if the share expiration date is not present then I will give it minimum expiration.

Reply:
The reason I am not touching existing shares, as they were already approved before and this would create a problems in which the user would have to be informed and also that the requester user has to keep in mind changes in the share expiration.